ros2bag/ros2bag/verb/record.py

```diff
@@ -67,9 +67,11 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
                   'the bag will split at whichever threshold is reached first.'
         )
         parser.add_argument(
-            '--max-cache-size', type=int, default=0,
-            help='maximum amount of messages to hold in cache before writing to disk. '
-                 'Default it is zero, writing every message directly to disk.'
+            '--max-cache-size', type=int, default=1024*1024,
```

Comment:
Does the default need to be changed [here](https://github.com/ros2/rosbag2/blob/19946e100dc19ae631966cd61fe1bb1d88c7080d/rosbag2_cpp/include/rosbag2_cpp/storage_options.hpp#L40) as well? There a few other places when searching for `max_cache_size`.

Reply:
Does the data structure encapsulating options care about the default value that the command line arg should use? In this case from the comments on the structure it seems like default values are defined inside it. I am not sure about this honestly.

Reply:
Perhaps yes for consistency. Otherwise, when someone inspects the code that may not have the same context as us, how does one infer the intended default unless stepping through the git blame process?

Also would note the new default in the argparse help string.

Reply:
Was just looking into the  code and I agree with you. In that case, I am going to split this PR into 2. Leave this one to just move from using number of messages in cache to message size and another with changes to the default one.

> Also would note the new default in the argparse help string.

I have updated the help string [here](https://github.com/ros2/rosbag2/pull/530/files/0469bfbd5f31f8c161ec1bd1fec70f7d1268d67f#diff-c3b61a120c8901c73b21f179439f4010R72)? Or do you mean something else?

Reply:
> I have updated the help string here? Or do you mean something else?

You're right: sorry, my mistake!